<?xml version="1.0"?>

<launch>
    <arg name="cbf_state_topic" />
    <arg name="cbf_safe_control_topic" />
    <arg name="cbf_nominal_control_topic" />
    <arg name="vf_update_topic" />
    <arg name="actuation_update_topic" />
    <arg name="disturbance_update_topic" />
    <arg name="obstacle_update_topic" />
    <arg name="safety_filter_active" />
    <node name="refine_cbf"
        pkg="refinecbf_ros"
        type="refine_cbf_node.py"
        output="screen"
        ns="safety_filter">
        <param name="topics/state" value="$(arg cbf_state_topic)" />
        <param name="topics/filtered_control" value="$(arg cbf_safe_control_topic)" />
        <param name="topics/nominal_control" value="$(arg cbf_nominal_control_topic)" />
        <param name="topics/actuation_update" value="$(arg actuation_update_topic)" />
        <param name="topics/vf_update" value="$(arg vf_update_topic)" />
        <param name="safety_filter_active" value="$(arg safety_filter_active)" />
    </node>
    <group if="$(eval arg('safety_filter_active') == 1)">
        <node name="hj_reachability"
            pkg="refinecbf_ros"
            type="hj_reachability_node.py"
            output="screen"
            ns="safety_filter">
            <param name="topics/disturbance_update" value="$(arg disturbance_update_topic)" />
            <param name="topics/actuation_update" value="$(arg actuation_update_topic)" />
            <param name="topics/obstacle_update" value="$(arg obstacle_update_topic)" />
            <param name="topics/vf_update" value="$(arg vf_update_topic)" />
        </node>
    </group>

</launch>